Source/Scene/ParticleSystem.js

```diff
@@ -0,0 +1,295 @@
+        this.bursts = defaultValue(options.bursts, null);
+
+        this.loop = defaultValue(options.loop, true);
+
+        var speed = defaultValue(options.speed, undefined);
+        if (speed) {
+            this.minSpeed = speed;
+            this.maxSpeed = speed;
+        }
+        else {
+            this.minSpeed = defaultValue(options.minSpeed, 1.0);
+            this.maxSpeed = defaultValue(options.maxSpeed, 1.0);
+        }
+
+        var life = defaultValue(options.life, undefined);
+        if (life) {
+            this.minLife = life;
+            this.maxLife = life;
+        }
+        else {
+            this.minLife = defaultValue(options.minLife, 5.0);
+            this.maxLife = defaultValue(options.maxLife, 5.0);
+        }
+
+        var mass = defaultValue(options.mass, undefined);
+        if (mass) {
+            this.minMass = mass;
+            this.maxMass = mass;
+        }
+        else {
+            this.minMass = defaultValue(options.minMass, 1.0);
+            this.maxMass = defaultValue(options.maxMass, 1.0);
+        }
+
+        this.image = defaultValue(options.image, null);
+
+        var width = defaultValue(options.width, undefined);
+        if (width) {
+            this.minWidth = width;
+            this.maxWidth = width;
+        }
+        else {
+            this.minWidth = defaultValue(options.minWidth, 1.0);
+            this.maxWidth = defaultValue(options.maxWidth, 1.0);
+        }
+
+        var height = defaultValue(options.height, undefined);
+        if (height) {
+            this.minHeight = height;
+            this.maxHeight = height;
+        }
+        else {
+            this.minHeight = defaultValue(options.minHeight, 1.0);
+            this.maxHeight = defaultValue(options.maxHeight, 1.0);
+        }
+
+        this.lifeTime = defaultValue(options.lifeTime, Number.MAX_VALUE);
+
+        this.complete = new Event();
+        this.isComplete = false;
+
+        this.show = defaultValue(options.show, true);
+
+        this.carryOver = 0.0;
+        this.currentTime = 0.0;
+        this._billboardCollection = undefined;
+        this._previousTime = null;
+    };
+
+    function removeBillboard(system, particle) {
+        system._billboardCollection.remove(particle._billboard);
+    }
+
+    function random(a, b) {
+        return CesiumMath.nextRandomNumber() * (b - a) + a;
+    }
+
+    function updateBillboard(system, particle) {
+        var billboard = particle._billboard;
+        if (!defined(billboard)) {
+            billboard = particle._billboard = system._billboardCollection.add({
+                image: particle.image
+            });
+        }
+        billboard.width = particle.size.x;
+        billboard.height = particle.size.y;
+        billboard.position = particle.position;
+
+        // Update the color
+        var r = particle.startColor.red + particle.normalizedAge * (particle.endColor.red - particle.startColor.red);
```

Comment:
Would [lerp](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Math.js#L311) work for these?